lib/multibuild/index.ts

```diff
@@ -128,27 +129,18 @@ export async function fromImageDescriptors(
 							newHeader.name = relative;
 							task.buildStream!.entry(newHeader, buf);
 						});
-					})
-					.then(() => {
 						next();
-						return null;
-					})
-					.catch((e) => {
-						if (e instanceof ContractError) {
-							reject(e);
-							return;
-						}
-						reject(new TarError(e));
-					});
-			} else {
-				TarUtils.drainStream(entryStream)
-					.then(() => {
-						next();
-					})
-					.catch((e) => {
-						reject(new TarError(e));
-					});
-			}
+					} else {
+						await TarUtils.drainStream(entryStream);
+					}
```

Comment:
I think there is a missing `next();` for the  `else`/`drainStream` case.
```suggestion
						next();
					}
```